### ISSUE TITLE
Delay whilst scrolling on widget 

### ISSUE THREAD

**Issue by adamosborne83** [jurialmunkey/skin.arctic.fuse.2#345]
### Skin section

Widgets

### Current Behavior

Hi J, 
I'm away from my tv at moment, so will send one of my famous videos if need be. But, has something changed since 2 4.0? Just when I scroll it takes a second or so for the background image to show up.

I know if nothing to do with Cache, as I have a program that caches all images in my library in the texture.db.

Also, when I go back to 2.4.0 the issue seems to go.

As always, thinking it might be size of my librarybsome how, as I don't notice this on my firestick in the bedroom.

### Expected Behavior

Same as 2.4.0

### Steps To Reproduce

Just happens.

### Screenshots and Additional Info

Will upload video later if you need.

### Checklist

- [X] I have made a separate issue for each bug that I am reporting.
- [X] I have searched the issues and this bug has not been reported.
- [X] I am using the most recent skin version.
- [X] I have confirmed that this bug does not occur in the default skin.

**Comment by jurialmunkey** [comment]
No there shouldn't be any changes impacting background artwork between 2.4.0 and 2.4.3

The last big change on that front was *in* 2.4.0 to how the TMDbHelper service is handled. So if you were saying you were going back *before* 2.4.0 that'd be a different story -- but nothing major *since* 2.4.0, just small tweaks and fixes.

**Comment by adamosborne83** [comment]
> No there shouldn't be any changes impacting background artwork between 2.4.0 and 2.4.3
> 
> The last big change on that front was *in* 2.4.0 to how the TMDbHelper service is handled. So if you were saying you were going back *before* 2.4.0 that'd be a different story -- but nothing major *since* 2.4.0, just small tweaks and fixes.

Sorry, I'm on 2.3.2.

Downloaded the zip for it.

**Comment by jurialmunkey** [comment]
This is the price of stability...

Either i use the experimental approach and hear about crashes or I use the stable approach and hear about performance.

Disable readahead caching in tmdbhelper other settings. Press OK and then restart kodi. Disabling readahead will avoid the service being busy looking ahead so that it will be quicker to respond.

**Comment by adamosborne83** [comment]
https://github.com/user-attachments/assets/ca4c4750-4991-40d5-a30f-ab02733135b5

It can sometimes be a good second or 2, but if it stops my crashes.....

**Comment by adamosborne83** [comment]
Enabling fake blur had a massive impact, now loads instantly, clearlogo is still a second or so delay, but, cannot have it all 😀

**Comment by adamosborne83** [comment]
I still crash though lmao.

**Comment by jurialmunkey** [comment]
> I still crash though lmao.

This is likely something unrelated to tmdbhelper then, because window property method is very safe.

**Comment by adamosborne83** [comment]
> > I still crash though lmao.
> 
> This is likely something unrelated to tmdbhelper then, because window property method is very safe. 

Thanks J, I'm going to play between 2.3.1, and your new releases....

Unrelated, I've turned off HEVC on the addon I use that seems to help stability.

**Comment by adamosborne83** [comment]
Maybe try a 'fake' clearlogo to speed it all up 😉

**Comment by jurialmunkey** [comment]
> 
> Thanks J, I'm going to play between 2.3.1, and your new releases....
> 

Please try latest version of skin `v2.4.4` with latest version of TMDbHelper `v5.4.0` with real blur + crop enabled. 


> I know if nothing to do with Cache, as I have a program that caches all images in my library in the texture.db.

Depending on how this utility works, it could be making matters significantly worse because the images for blur/crop are stored separately from the library.

If this utility is clearing out the texture cache before it updates it with library images, then this will be clearing out all blur and crop images. That will make texture performance significantly worse because Kodi will be need to re-cache any images not in your library (including the blur and crop images). 

The appearance of the big blank space in the logo area makes me think this is what is occurring. If there were no logo set then you would get a text based title. If the area is blank then what you are waiting on is Kodi to cache the image.

The utility must be only additive (i.e. it never clears anything only adds missing entries) to not impact texture performance.

**Comment by adamosborne83** [comment]
> > 
> > Thanks J, I'm going to play between 2.3.1, and your new releases....
> > 
> 
> Please try latest version of skin `v2.4.4` with latest version of TMDbHelper `v5.4.0` with real blur + crop enabled. 
> 
> 
> > I know if nothing to do with Cache, as I have a program that caches all images in my library in the texture.db.
> 
> Depending on how this utility works, it could be making matters significantly worse because the images for blur/crop are stored separately from the library.
> 
> If this utility is clearing out the texture cache before it updates it with library images, then this will be clearing out all blur and crop images. That will make texture performance significantly worse because Kodi will be need to re-cache any images not in your library (including the blur and crop images). 
> 
> The appearance of the big blank space in the logo area makes me think this is what is occurring. If there were no logo set then you would get a text based title. If the area is blank then what you are waiting on is Kodi to cache the image.
> 
> The utility must be only additive (i.e. it never clears anything only adds missing entries) to not impact texture performance.
> 

Good Morning J,

Yes your latest updates help significantly. See video, clearlogo is instant now, and little no delay.
Thank you for this, I hope you didn't have to compromise of stability to sort this for me.

https://github.com/user-attachments/assets/f30472b6-b0d1-4e49-9183-15a93eaf0946

Also, in reference to my cache program, that i use.

My understanding is it reads to see if the images are cached already, and if not will download to cache them, it does not delete or clean anything, just adds.

See photos below for how it's used.

![firetv_2024-11-10_08_28_21.jpg](https://github.com/user-attachments/assets/580d1743-ce5e-4b78-956b-88b487e99e34)

![firetv_2024-11-10_08_28_36.jpg](https://github.com/user-attachments/assets/59288e01-830d-4fa4-b418-60c40bf7bab1)

![firetv_2024-11-10_08_29_03.jpg](https://github.com/user-attachments/assets/d8157dfe-c363-4676-9a87-191c30d8ee98)

![firetv_2024-11-10_08_30_07.jpg](https://github.com/user-attachments/assets/a6bea2ab-a92e-40da-8205-0ec1d3726948)

![firetv_2024-11-10_08_28_44.jpg](https://github.com/user-attachments/assets/2dac1827-76d8-4c2e-bed1-9ca6beb24324)

![firetv_2024-11-10_08_31_33.jpg](https://github.com/user-attachments/assets/737c663f-75ef-4c73-9ea1-21b77adb799b)

![firetv_2024-11-10_08_29_09.jpg](https://github.com/user-attachments/assets/3d0d101d-e496-4fa3-948b-ac3734bdc9d5)

**Comment by adamosborne83** [comment]
The above photos arnt in order but you get the idea. 17 images aren't cached so it will download those, then after the process it will then go again on all 22k images.

**Comment by jurialmunkey** [comment]
> Yes your latest updates help significantly. See video, clearlogo is instant now, and little no delay.
Thank you for this, I hope you didn't have to compromise of stability to sort this for me.

Great to hear!

And no compromise. It's still using the same "safe" window property approach. 

The difference now is that the image manipulations go in their own separate loop so that they don't get stuck in the queue behind the thread handling the detailed information lookup (it's a little more complicated than that but you get the idea - basically has its own separate fast pass queue).

Been meaning to make that change for a while. Was fairly sure it'd perform better while side stepping the underlying Kodi bugs in the more "experimental" method I was using before. It was more about finding the motivation, especially when the other approach is so convenient from a skinning perspective.

> My understanding is it reads to see if the images are cached already, and if not will download to cache them, it does not delete or clean anything, just adds.

Ah right, sort of similar to the Artwork Dump addon.

Yeah if it doesn't clean/delete any entries then it wouldn't be any cause for concern.

**Comment by adamosborne83** [comment]
Thank you for this.
